bentoml/_internal/bento/build_config.py

```diff
@@ -58,33 +59,74 @@ def _convert_python_version(py_version: t.Optional[str]) -> t.Optional[str]:
     if not isinstance(py_version, str):
         py_version = str(py_version)
 
-    match = re.match(r"^(\d+)\.(\d+)(?:\.\w+)?", py_version)
+    match = re.match(r"^(\d{1})\.(\d{,2})(?:\.\w+)?$", py_version)
     if match is None:
         raise InvalidArgument(
             f'Invalid build option: docker.python_version="{py_version}", python '
             f"version must follow standard python semver format, e.g. 3.7.10 ",
         )
     major, minor = match.groups()
-    return f"{major}.{minor}"
+    target_python_version = f"{major}.{minor}"
+    if target_python_version != py_version:
+        logger.warning(
+            "BentoML will install the latest python%s instead of the specified version %s. To use the exact python version, use a custom docker base image. See https://docs.bentoml.org/en/latest/concepts/bento.html#custom-base-image-advanced",
+            target_python_version,
+            py_version,
+        )
+    return target_python_version
 
 
 def _convert_cuda_version(
     cuda_version: t.Optional[t.Union[str, int]]
 ) -> t.Optional[str]:
-    if cuda_version is None or cuda_version == "":
+    if cuda_version is None or cuda_version == "" or cuda_version == "None":
         return None
 
-    if not isinstance(cuda_version, str):
+    if isinstance(cuda_version, int):
         cuda_version = str(cuda_version)
 
     if cuda_version == "default":
-        cuda_version = DEFAULT_CUDA_VERSION
+        return DEFAULT_CUDA_VERSION
+
+    if cuda_version in ALLOWED_CUDA_VERSION_ARGS:
+        _cuda_version = ALLOWED_CUDA_VERSION_ARGS[cuda_version]
+        if _cuda_version in SUPPORTED_CUDA_VERSIONS:
+            return _cuda_version
 
-    match = re.match(r"^(\d+)$", cuda_version)
-    if match is not None:
-        cuda_version = SUPPORTED_CUDA_VERSIONS[cuda_version]
+    raise BentoMLException(
+        f'Unsupported cuda version: "{cuda_version}". Supported cuda versions are {list(ALLOWED_CUDA_VERSION_ARGS.keys())}'
+    )
 
-    return cuda_version
+
+def _convert_env(
+    env: t.Optional[t.Union[str, t.List[str], t.Dict[str, str]]]
+) -> t.Optional[t.Dict[str, str]]:
+    if env is None:
+        return None
+
+    if isinstance(env, str):
+        logger.debug("Reading dot env file '%s' specified in config", env)
+        return dict(dotenv_values(env))
```

Comment:
dotenv_values returns an OrderedDict and it causes issues with pyyaml - thus converting it to `dict`